test.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import unittest
+from inspect import cleandoc
+from expr_checker import *
+
+class Dangerous:
+    pass
+
+class Good:
+    a = "hi"
+    evil = Dangerous()
+
+    _secret_stuff = "1+1=3"
+
+    def tell_me_hi(self):
+        return self.a
+    
+    def __enter__(self):
+        return Good()
+
+    def __exit__(self, exec_type, exec_value, traceback):
+        pass
+
+
+class ReadOnlyObject:
+    def __init__(self, x):
+        self.set_x_value(x)
+
+    def set_x_value(self, x):
+        self.x = x
+
+
+_WHITELIST = {str, int, type(None), Good, type(range(0))}
+_ALLOWED_ATTR = {"a", "x", "tell_me_hi", "set_x_value"}
+
+
+class TestFuncChecker(unittest.TestCase):
+    def test_function_call(self):
+        def abc(a, b, c):
+            return f"A: {a}, B: {b}, C: {c}"
+
+        self.assertEqual(
+            eval(expr_checker("abc('aaa', 'bbb', 'ccc')",
+                 _WHITELIST, _ALLOWED_ATTR, {})),
+            "A: aaa, B: bbb, C: ccc"
+        )
+
+        with self.assertRaises(ValueError):
+            eval(expr_checker("abc(a='aaa', c='ccc', b=Dangerous())",
```

Comment:
can you create a `eval` function in a different file, import it and use it instead of the classic eval ?

This function should eval the code received in argument, and clean the globals / locals (to ensure that you add `test`, `check_type`,...) in the generated python code (cf our safe_eval code)

```python
_BUILTINS = {
    'True': True,
    'False': False,
    'None': None,
    'bytes': bytes,
    'str': str,
    'unicode': str,
    'bool': bool,
    'int': int,
    'float': float,
    'enumerate': enumerate,
    'dict': dict,
    'list': list,
    'tuple': tuple,
    'map': map,
    'abs': abs,
    'min': min,
    'max': max,
    'sum': sum,
    'filter': filter,
    'sorted': sorted,
    'round': round,
    'len': len,
    'repr': repr,
    'set': set,
    'all': all,
    'any': any,
    'ord': ord,
    'chr': chr,
    'divmod': divmod,
    'isinstance': isinstance,
    'range': range,
    'xrange': range,
    'zip': zip,
    'Exception': Exception,
    'print': print,
}


def eval(code):
    _globals = {'__builtins__': _BUILTINS}
    _locals = {}
    return exec(code, _globals, _locals)


_eval('print(1)')
```